src/NUnitEngine/nunit.engine/Internal/DomainDetailsBuilder.cs

```diff
@@ -0,0 +1,81 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace NUnit.Engine.Internal
+{
+    /// <summary>
+    /// DomainDetailsBuilder provides human readable information on 
+    /// an AppDomain, to assist with debugging.
+    /// </summary>
+    internal static class DomainDetailsBuilder
+    {
+        /// <summary>
+        /// Get human readable string containing details of AppDomain.
+        /// </summary>
+        /// <param name="domain">AppDomain to get details on.</param>
+        /// <param name="errMsg">An optional overall error message e.g. The messaged from an AppDomainUnload exception.</param>
+        public static string DetailsFor(AppDomain domain, string errMsg = null)
+        {
+            var sb = new StringBuilder();
+            if (errMsg != null) sb.AppendLine(errMsg);
+
+            try
+            {
+                sb.AppendLine($"AppDomain Name: {domain.FriendlyName}");
+                sb.AppendLine($"AppDomain Base Directory: {domain.BaseDirectory}");
+
+                var reflectionLoadedAssemblies = new List<Assembly>(domain.ReflectionOnlyGetAssemblies());
+
+                if (reflectionLoadedAssemblies.Count != 0)
+                {
+                    sb.AppendLine("--- Assemblies loaded in current AppDomain via reflection ---");
+                    foreach (var assembly in reflectionLoadedAssemblies)
+                        WriteAssemblyInformation(sb, assembly);
+                }
+            }
+            catch (AppDomainUnloadedException)
+            {
+                sb.AppendLine("App Domain was unloaded before all details could be read.");
```

Comment:
In all other places we use the term "AppDomain" (i.e. in one word)

Reply:
Thanks for this comment. I decided to refer to what MS do in their docs - they write out 'application domain' in full, when referring to the concept. Based on that, I went through and changed all the public string to that convention instead. 🙂